subsys/bluetooth/host/conn.c

```diff
@@ -53,8 +53,13 @@ NET_BUF_POOL_DEFINE(acl_tx_pool, CONFIG_BT_L2CAP_TX_BUF_COUNT,
  * are queued up in the TX queue. In such a situation, trying to allocate
  * another buffer from the acl_tx_pool would result in a deadlock.
  */
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
+NET_BUF_POOL_DEFINE(frag_pool, CONFIG_BT_L2CAP_TX_FRAG_COUNT, FRAG_SIZE,
+		    BT_BUF_USER_DATA_MIN, NULL);
+#else
 NET_BUF_POOL_FIXED_DEFINE(frag_pool, CONFIG_BT_L2CAP_TX_FRAG_COUNT, FRAG_SIZE,
 			  NULL);
+#endif /* CONFIG_BT_L2CAP_DYNAMIC_CHANNEL */
```

Comment:
What's the difference here? `NET_BUF_POOL_DEFINE()` is the same as `NET_BUF_POOL_FIXED_DEFINE()` in fact, you shouldn't use `NET_BUF_POOL_DEFINE()` anymore as it's essentially deprecated (even though it might not be formally marked so).

Reply:
It was just for setting user_data size, the fixed does seem to take the user_data size.

Reply:
Take a look at the actual definition. All it does with it is a build assert. Ensuring the BT_BUF_USER_DATA_MIN is already covered by `range 8 65535 if BT` in `subsys/net/Kconfig`

Reply:
If that still wasn't clear: since quite a while ago all net_buf objects share the same user data size that gets set using `CONFIG_NET_BUF_USER_DATA_SIZE`. I.e. you can't set it per-pool

Reply:
Ok now I realize that it doesn't actually assume a 0 size with fixed version, will fix that.

Reply:
Fixed.